awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,25 @@
+/** @odoo-module **/
+
+import { Component, useState } from "@odoo/owl";
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter";
+
+    static props = {
+        callback: {
+            type: Function,
+            optional: true,
+        }
+    }
+
+    setup() {
+        this.count = useState({ value: 0 });
+    }
+
+    increment() {
+        this.count.value += 1;
+        if (this.props.callback) {
+            this.props.callback();
+        }
```

Comment:
```suggestion
        this.props.callback?.();
```
This is an easier way to write down the check before the call if exists.